src/test/java/com/devonfw/sample/archunit/ComponentRuleTest.java

```diff
@@ -0,0 +1,263 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+
+import static com.tngtech.archunit.base.DescribedPredicate.alwaysTrue;
+import static com.tngtech.archunit.core.domain.properties.HasName.Predicates.nameMatching;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class ComponentRuleTest {
+
+    /**
+     * verifying that the service layer of one component does not depend on the
+     * service layer of
+     * another component.
+     */
+    @ArchTest
+    public static final ArchRule C3_no_service_layer_depends_on_service_layer_of_another_component = slices()
+            .matching("..archunit.(*).service..")
+            .namingSlices("$1 service").should().notDependOnEachOther()
+            .ignoreDependency(alwaysTrue(), nameMatching(".*general.*"))
```

Comment:
This will also match to e.g. `com.devonfw.sample.archunit.component.service.foogeneral`.
I guess applying `StructuredPackage` to all these places will solve lots of problems.

Reply:
~~Adjusted the ignoreDependency like your [proposal for] (https://github.com/devonfw-sample/archunit/pull/37#discussion_r1135177629)~~

~~`.matching("..(*).service..").namingSlices("$1 service").should().notDependOnEachOther()
      .ignoreDependency(alwaysTrue(), nameMatching(".general.(common|service|logic|dataaccess|batch|client).."))`~~

Reimplemented rules C3 and C6 due to the erroneous use of ignoreDependency().
A custom ArchCondition combined with `PackageStructure `is utilized now to enfore rules C3 and C6.